src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -799,6 +799,44 @@ BOOL CEEInfo::isValidStringRef (
     return result;
 }
 
+int CEEInfo::getStringLength (
+        CORINFO_MODULE_HANDLE       moduleHnd,
+        mdToken                     metaTOK)
+{
+    CONTRACTL{
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    int length = 0;
+    Module* module = GetModule(moduleHnd);
+
+    JIT_TO_EE_TRANSITION();
+
+    if (IsDynamicScope(moduleHnd))
```

Comment:
What it would take to obtain the string characters as well in this case? So that the new jit interface function can also return the characters?

Reply:
@mikedn do you have a usecase for it? I remember I tried to optimize `Equals(string,sting)` and made a JIT method `getString` but jit-diff was almost empty 🙁

Reply:
Well, the point is that sooner or later someone will come up with an use case and then we'll need yet another jit interface change. So if it's not hard to return the characters it would perhaps be good to just do it now. Though I suspect that the dynamic module case may be tricky so that's why I'm asking.

Reply:
ok, will try (in a separate commit)

Reply:
@mikedn extend the API to return `const wchar_t*` (and renamed to `GetStringLiteral`) in the most recent commit.
I don't need the actual string in my PR so I use it for JITDUMP [here](https://github.com/dotnet/runtime/blob/286d7a976bb8d529f363b8df866df393fcdce7c2/src/coreclr/src/jit/importer.cpp#L3640-L3643).
And yeah I am not sure in the dynamic module implementation yet.